From 0bda19d1237ef490cb6005f9946ccd897ec61417 Mon Sep 17 00:00:00 2001
From: "richard.li" <richard.li@linux.alibaba.com>
Date: Fri, 28 Oct 2022 15:54:41 +0800
Subject: [PATCH] fpga: dfl: add vendor IDs and device IDs for ALIBABA F5 cards

This patch adds the approved PCI Express Vendor IDs and Device IDs for the
ALIBABA cards.

Signed-off-by: richard.li <richard.li@linux.alibaba.com>
---
 drivers/fpga/dfl-pci.c  | 2 ++
 include/linux/pci_ids.h | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/fpga/dfl-pci.c b/drivers/fpga/dfl-pci.c
index 80cac3a5f..e2128d7c7 100644
--- a/drivers/fpga/dfl-pci.c
+++ b/drivers/fpga/dfl-pci.c
@@ -79,6 +79,7 @@
 #define PCIE_DEVICE_ID_SILICOM_PAC_N5013	0x1002
 #define PCIE_DEVICE_ID_SILICOM_PAC_N5014	0x1003
 #define PCIE_DEVICE_ID_INTEL_DFL		0xbcce
+#define PCIE_DEVICE_ID_ALIBABA_F5		0x8103
 /* PCI Subdevice ID for PCIE_DEVICE_ID_INTEL_DFL */
 #define PCIE_SUBDEVICE_ID_INTEL_D5005		0x138d
 #define PCIE_SUBDEVICE_ID_INTEL_N6000		0x1770
@@ -120,6 +121,7 @@
 			PCI_VENDOR_ID_INTEL, PCIE_SUBDEVICE_ID_INTEL_C6100),},
 	{PCI_DEVICE_SUB(PCI_VENDOR_ID_INTEL, PCIE_DEVICE_ID_INTEL_DFL_VF,
 			PCI_VENDOR_ID_INTEL, PCIE_SUBDEVICE_ID_INTEL_C6100),},
+	{PCI_DEVICE(PCI_VENDOR_ID_ALIBABA, PCIE_DEVICE_ID_ALIBABA_F5),},
 	{0,}
 };
 MODULE_DEVICE_TABLE(pci, cci_pcie_id_tbl);
diff --git a/include/linux/pci_ids.h b/include/linux/pci_ids.h
index 0a85ff5c8..41161e4b1 100644
--- a/include/linux/pci_ids.h
+++ b/include/linux/pci_ids.h
@@ -1381,6 +1381,8 @@
 #define PCI_DEVICE_ID_IMS_TT128		0x9128
 #define PCI_DEVICE_ID_IMS_TT3D		0x9135
 
+#define PCI_VENDOR_ID_ALIBABA		0x1ded
+
 #define PCI_VENDOR_ID_AMCC		0x10e8
 #define PCI_VENDOR_ID_AMPERE		0x1def
 